usaspending_api/recipient/delta_models/recipient_lookup.py

```diff
@@ -1,30 +1,686 @@
+    "source": {"delta": "STRING NOT NULL", "postgres": "TEXT NOT NULL"},
+    "update_date": {"delta": "TIMESTAMP NOT NULL", "postgres": "TIMESTAMP NOT NULL"},
+    "uei": {"delta": "STRING", "postgres": "TEXT"},
+    "parent_uei": {"delta": "STRING", "postgres": "TEXT"},
+}
+
 RECIPIENT_LOOKUP_COLUMNS = {
-    "id": "LONG NOT NULL",
-    "recipient_hash": "STRING",
-    "legal_business_name": "STRING",
-    "duns": "STRING",
-    "address_line_1": "STRING",
-    "address_line_2": "STRING",
-    "business_types_codes": "ARRAY<STRING>",
-    "city": "STRING",
-    "congressional_district": "STRING",
-    "country_code": "STRING",
-    "parent_duns": "STRING",
-    "parent_legal_business_name": "STRING",
-    "state": "STRING",
-    "zip4": "STRING",
-    "zip5": "STRING",
-    "alternate_names": "ARRAY<STRING>",
-    "source": "STRING NOT NULL",
-    "update_date": "TIMESTAMP NOT NULL",
-    "uei": "STRING",
-    "parent_uei": "STRING",
+    "id": {"delta": "LONG NOT NULL", "postgres": "BIGINT NOT NULL"},
+    **RECIPIENT_LOOKUP_COLUMNS_WITHOUT_ID,
+}
+
+RECIPIENT_LOOKUP_DELTA_COLUMNS = {k: v["delta"] for k, v in RECIPIENT_LOOKUP_COLUMNS.items()}
+RECIPIENT_LOOKUP_POSTGRES_COLUMNS = {k: v["postgres"] for k, v in RECIPIENT_LOOKUP_COLUMNS.items()}
+
+TEMP_RECIPIENT_LOOKUP_COLUMNS = {
+    "duns_recipient_hash": "STRING",
+    "row_num_union": "INTEGER",
+    **{k: v["delta"] for k, v in RECIPIENT_LOOKUP_COLUMNS_WITHOUT_ID.items()},
 }
 
-recipient_lookup_sql_string = rf"""
+recipient_lookup_create_sql_string = rf"""
     CREATE OR REPLACE TABLE {{DESTINATION_TABLE}} (
-        {", ".join([f'{key} {val}' for key, val in RECIPIENT_LOOKUP_COLUMNS.items()])}
+        {", ".join([f'{key} {val}' for key, val in RECIPIENT_LOOKUP_DELTA_COLUMNS.items()])}
     )
     USING DELTA
     LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
     """
+
+recipient_lookup_load_sql_string_list = [
+    # -----
+    # Creation of a temporary view used to reference recipient information from transactions
+    # -----
+    r"""
+    CREATE OR REPLACE TEMPORARY VIEW temporary_transaction_recipients_view AS (
+        SELECT
+            tn.transaction_unique_id,
+            tn.is_fpds,
+            REGEXP_REPLACE(
+                MD5(UPPER(
+                    CASE
+                        WHEN COALESCE(fpds.awardee_or_recipient_uei, fabs.uei) IS NOT NULL THEN CONCAT('uei-', COALESCE(fpds.awardee_or_recipient_uei, fabs.uei))
+                        WHEN COALESCE(fpds.awardee_or_recipient_uniqu, fabs.awardee_or_recipient_uniqu) IS NOT NULL THEN CONCAT('duns-', COALESCE(fpds.awardee_or_recipient_uniqu, fabs.awardee_or_recipient_uniqu))
+                        ELSE CONCAT('name-', COALESCE(fpds.awardee_or_recipient_legal, fabs.awardee_or_recipient_legal)) END
+                )),
+                '^(\.{{8}})(\.{{4}})(\.{{4}})(\.{{4}})(\.{{12}})$',
+                '\$1-\$2-\$3-\$4-\$5'
```

Comment:
Added a comment to clarify. Essentially this is how we can format MD5 to be a "UUID" as what would be expected in Postgres without the need for a UDF since all UUIDs should have the same formet.